### DISCUSSION TITLE
1.0 버젼 이후 개선해보고 싶은 사항을 남겨봅시다 🌱

### DISCUSSION THREAD

**Discussion by Kim0914** [woowacourse-teams/2023-pium#307] [개발 논의]
## 남겨봐요 😆

**Comment by Kim0914** [comment]
* 사진과 글자 간격을 좀 더 넒게 하면 좋을 것 같아요 ㅎㅎ
<img width="410" alt="image" src="https://github.com/woowacourse-teams/2023-pium/assets/68818952/503dc8ec-ae0d-4d39-b471-4c4a7e313992">

**Comment by Choi-JJunho** [comment]
- confirm 디자인 확인
- 토스트 위치 변경 (상단으로)
- 식물 사전에서 바로 등록하는 버튼 만들기
- 내 식물 -> 식물 사전 -> 내 식물로 다시 이동 가능한 버튼?

**Comment by Choi-JJunho** [comment]
서비스 상에서 사전 식물을 바로 등록할 수 있는 ADMIN 기능이 있으면 좋겠어요

**Comment by Choi-JJunho** [comment]
검색할 때 자음만 쳤을 때도 검색이 수행되었으면 좋겠어요!

**Comment by Choi-JJunho** [comment]
무중단 배포를 수행했으면 좋겠어요!
생각나는 고려 사항은 다음과 같습니다!

- 로드밸런싱
  - 두개의 WAS를 이용해 무중단 배포 환경을 구성한다면 배포를 수행하고있는 서버(요청을 받지 못하는 서버)에는 요청이 가지 않도록 구성해야합니다.
- 포트포워딩
  - 하나의 서버로 두개의 WAS를 띄운다고 가정하면 서로 다른 포트로 띄워야합니다.
- 세션 클러스터링
  - 현재 스프링 컨텍스트에 세션 정보가 저장되는데 이 정보를 다른 WAS에서도 공유받아야합니다.
  - 키워드: Session Clustering, Sticky Session, Redis Session Clustering

**Comment by Choi-JJunho** [comment]
현재 서비스 상으로 대용량 트래픽을 경험하지 못하므로 부하테스트를 도입해서 쿼리 속도측정 및 개선과정을 경험해봤으면 좋겠어요!

부하테스트 툴 목록

- [JMeter](https://jmeter.apache.org/)
- [개틀링](https://gatling.io/)
- [K6](https://k6.io/)
- [Chaos Monkey](https://netflix.github.io/chaosmonkey/)
- [Ngrinder](https://naver.github.io/ngrinder/)

**Comment by Kim0914** [comment]
Jacoco와 같은 도구를 이용해 테스트 커버리지 제한을 둬도 좋을 것 같아요 !

**Comment by Choi-JJunho** [comment]
다음 discussion 내용과 함께 close 하겠습니다.
- https://github.com/woowacourse-teams/2023-pium/discussions/317
